src/Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -221,12 +222,13 @@ public class ConsensusLoop : IConsensusLoop
             while (true)
             {
                 this.Tip = this.Chain.GetBlock(utxoHash);
-                if (this.Tip != null)
+
+                bool blockStoreDisabled = this.chainState.BlockStoreTip == null;
+                if ((this.Tip != null) && (blockStoreDisabled || (this.chainState.BlockStoreTip.Height >= this.Tip.Height)))
```

Comment:
since one part of the conditional has been given a variable, it would be helpful if the other two parts also were so that it is easier to read

Reply:
I don't think having a separated variable for `this.Tip != null` is a good idea. It will just add an additional line without adding more reading simplicity.

Reply:
Yep, ok I accept that, and if you did the last part there is a null reference, so yeah, not much you could do without wrapping the whole boolean in a meaningful name or method.

Reply:
This condition `bool blockStoreDisabled = this.chainState.BlockStoreTip == null;` can go outside the while, infact you don't even need this while if store is disabled
just this line
`this.Tip = this.Chain.GetBlock(utxoHash);`

Reply:
>infact you don't even need this while if store is disabled

You still need- when there is no chainblock for consensus tip (Tip == null) we use this as well

Reply:
>This condition bool blockStoreDisabled = this.chainState.BlockStoreTip == null; can go outside the while,

Agreed. Fixed